lms/djangoapps/edraak_university/forms.py

```diff
@@ -0,0 +1,67 @@
+"""
+Forms for the University ID django app.
+"""
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+import re
+
+from openedx.core.djangoapps.user_api.accounts import NAME_MIN_LENGTH
+
+from edraak_university.models import UniversityID
+
+
+class UniversityIDForm(forms.ModelForm):
+    """
+    Form for the student view to enter their ID and section number, and change the full name.
+    """
+    full_name = forms.CharField(
+        label=_('Full Name in Arabic *'),
+        help_text=_('Enter your full name that you use at the university in Arabic.'),
+        required=True,
+        min_length=NAME_MIN_LENGTH,
+        max_length=50,  # Just a basic sanity check
+        error_messages={
+            'min_length': _('The name you have entered is too short, please double check.'),
+            'max_length': _('The student university ID you have entered is too long, please double check.'),
+        },
+    )
+
+    university_id = forms.RegexField(
+        regex=re.compile(r'^[0-9a-z-]+$', re.IGNORECASE),
+        label=_('Student University ID *'),
+        # TODO: Make ID format instruction course-variant, so coordinators can define it for each course.
+        help_text=_('Enter the full student university ID e.g. 201311318.'),
+        min_length=4,
+        max_length=50,
+        error_messages={
+            'invalid': _('The student university ID must only consist of numbers, letters and dashes.'),
+            'min_length': _('The student university ID you have entered is too short, please double check.'),
+            'max_length': _('The student university ID you have entered is too long, please double check.'),
+        },
+    )
+
+    section_number = forms.CharField(
```

Comment:
edX already has a cohort system. And integration is planned but still not there. So eventually this should be done later.